src/jit/lower.cpp

```diff
@@ -960,7 +960,16 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
 #endif // FEATURE_MULTIREG_ARGS
 #endif // not defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
         {
-            putArg = comp->gtNewOperNode(GT_PUTARG_REG, type, arg);
+#ifdef _TARGET_ARM_
+            if (comp->opts.compUseSoftFP && type == TYP_DOUBLE)
+            {
+                putArg = new (comp, GT_PUTARG_REG) GenTreeMultiRegOp(GT_PUTARG_REG, type, arg, nullptr);
```

Comment:
In order to reduce `#ifdef`s here, it would be good to add a `Compiler::gtNewPutArgReg()` method.